File: src/encoders/account-kind.ts
import { AccountKind } from "~/models";

export const encodeAccountKindToPath = (kind: AccountKind): string => {
  let name: string;

  switch (kind) {
    case AccountKind.STUDENT:
      name = "eleve";
      break;
    case AccountKind.PARENT:
      name = "parent";
      break;
    case AccountKind.TEACHER:
      name = "professeur";
      break;
  }

  return `mobile.${name}.html`;
};

File: library/src/commonMain/kotlin/encoders/account-kind.kt
package encoders

import models.AccountKind

fun encodeAccountKindToPath(kind: AccountKind): String {
    val name = when(kind) {
        AccountKind.STUDENT -> "eleve"
        AccountKind.PARENT -> "parent"
        AccountKind.TEACHER -> "professeur"
    }

    return "mobile.${name}.html"
}